documentation/modules/deploying/proc-deploy-migrate-kraft.adoc

```diff
@@ -0,0 +1,253 @@
+[source,shell]
+----
+NAME                     READY  STATUS   RESTARTS
+my-cluster-broker-0      1/1    Running  0
+my-cluster-broker-1      1/1    Running  0
+my-cluster-broker-2      1/1    Running  0
+my-cluster-controller-3  1/1    Running  0
+my-cluster-controller-4  1/1    Running  0
+my-cluster-controller-5  1/1    Running  0
+# ...
+----
+
+. Check the status of the migration:
++
+[source,shell]
+----
+kubectl get kafka my-cluster -n my-project -w
+----
++
+.Updates to the metadata state
+[source,shell]
+----
+NAME        ...  METADATA STATE   WARNINGS
+my-cluster  ...  Zookeeper
+my-cluster  ...  KRaftMigration
+my-cluster  ...  KRaftDualWriting
+----
++
+`METADATA STATE` shows the mechanism used to manage Kafka metadata and coordinate operations.
+At the start of the migration this is `ZooKeeper`.
++
+--
+* `ZooKeeper` is the initial state when metadata is only stored in ZooKeeper.
+* `KRaftMigration` is the state when the migration is in progress.
+* `KRaftDualWriting` is the state when the migration has ended and the cluster is working as a KRaft cluster. 
+Metadata is managed in Kafka but also replicated in ZooKeeper, so you can xref:proc-deploy-migrate-kraft-rollback-{context}[roll back from this point].
+--
++
+The migration status is also represented in the `status.KafkaMetadataState` property of the `Kafka` resource. 
+
+. Enable KRaft in the `Kafka` resource configuration by setting the `strimzi.io/kraft` annotation to `enabled`:
++
+[source,shell]
+----
+kubectl annotate kafka my-cluster strimzi.io/kraft: enabled
+----
++
+.Enabling KRaft migration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+  namespace: my-project
+  annotations:
+    strimzi.io/kraft: enabled 
+# ...
+----
+
+. Check the status of the move to full KRaft mode:
++
+[source,shell]
+----
+kubectl get kafka my-cluster -n my-project -w
+----
++
+.Updates to the metadata state
+[source,shell]
+----
+NAME        ...  METADATA STATE        WARNINGS
+my-cluster  ...  Zookeeper
+my-cluster  ...  KRaftMigration
+my-cluster  ...  KRaftDualWriting
+my-cluster  ...  KRaftPostMigration
+my-cluster  ...  KRaft                 True
+----
++
+`KRaftPostMigration` is the state (after the brokers have rolled) when KRaft mode is enabled and there is no ZooKeeper involvement, though ZooKeeper pods are still running.
+`KRaft` is the final state (after the controllers have rolled) when the KRaft migration has finalized.
++
+The warning (`True`) relates to ZooKeeper configuration being present in the `Kafka` resource configuration.
++
+. Clean up the deployment to remove Zookeeper configuration and resources.
+
+.. Remove `inter.broker.protocol.version`, `log.message.format.version`, and all `spec.zookeeper` configuration properties from the `Kafka` resource.
+.. Apply the changes to the `Kafka` resource configuration.
+.. Delete any resources related to the ZooKeeper deployment:
```

Comment:
Isn't this done automatically?

Reply:
I think Paolo was working towards that. Maybe we're there.

Reply:
Yes we are there now. All the ZooKeeper resources are automatically deleted now.

Reply:
Great. I've dropped this step